contracts/axelarnet-gateway/src/clients/external.rs

```diff
@@ -11,39 +12,47 @@ pub struct AxelarExecutableMsg {
     pub payload: HexBinary,
 }
 
-/// Crate-specific `ExecuteMsg` type wraps the `AxelarExecutableMsg` for the AxelarExecutable client.
+/// By convention, amplifier-compatible contracts must expose this `Execute` variant.
+/// Due to identical json serialization, we can imitate it here so the gateway can call it.
 #[cw_serde]
-pub enum AxelarExecutableExecuteMsg {
+pub enum ExecuteMsg {
     /// Execute the message at the destination contract with the corresponding payload.
     Execute(AxelarExecutableMsg),
 }
 
-impl<'a> From<client::Client<'a, AxelarExecutableExecuteMsg, ()>> for AxelarExecutableClient<'a> {
-    fn from(client: client::Client<'a, AxelarExecutableExecuteMsg, ()>) -> Self {
-        AxelarExecutableClient { client }
-    }
+pub struct CrossChainExecutor<'a> {
+    client: client::Client<'a, ExecuteMsg, ()>,
 }
 
-pub struct AxelarExecutableClient<'a> {
-    client: client::Client<'a, AxelarExecutableExecuteMsg, ()>,
-}
+impl<'a> CrossChainExecutor<'a> {
+    pub fn new(deps: Deps<'a>, destination: &str) -> error_stack::Result<Self, address::Error> {
+        let destination = address::validate_cosmwasm_address(deps.api, destination)?;
+        Ok(CrossChainExecutor {
+            client: client::Client::new(deps.querier, destination),
+        })
+    }
 
-impl<'a> AxelarExecutableClient<'a> {
-    pub fn execute(
+    pub fn prepare_execute_msg(
         &self,
         cc_id: CrossChainId,
         source_address: Address,
         payload: HexBinary,
     ) -> WasmMsg {
         self.client
-            .execute(&AxelarExecutableExecuteMsg::Execute(AxelarExecutableMsg {
+            .execute(&ExecuteMsg::Execute(AxelarExecutableMsg {
                 cc_id,
                 source_address,
                 payload,
             }))
     }
 }
 
+impl<'a> From<client::Client<'a, ExecuteMsg, ()>> for CrossChainExecutor<'a> {
```

Comment:
I like including `Client` suffix in the names to make it obvious it's a client

Reply:
that's the point though, it's not really a client.the only thing it does is prepare a message, there are no queries, and the message only works with other contracts by convention

Reply:
I suggested an alternate approach here that makes it more clear it's not a client:
https://github.com/axelarnetwork/axelar-amplifier/pull/583#discussion_r1729519076